<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>900</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Cheetah GUI</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="1" column="0">
     <layout class="QVBoxLayout" name="verticalLayout">
      <property name="sizeConstraint">
       <enum>QLayout::SetDefaultConstraint</enum>
      </property>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="sizeConstraint">
         <enum>QLayout::SetDefaultConstraint</enum>
        </property>
        <item>
         <widget class="QPushButton" name="button_refresh">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>120</width>
            <height>25</height>
           </size>
          </property>
          <property name="text">
           <string>Refresh</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="button_run_cheetah">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>120</width>
            <height>25</height>
           </size>
          </property>
          <property name="text">
           <string>Run Cheetah</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="button_run_streaming">
          <property name="text">
           <string>Run Cheetah+CrystFEL</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="button_kill_processing">
          <property name="text">
           <string>Kill processing</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="button_view_hits">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>120</width>
            <height>25</height>
           </size>
          </property>
          <property name="text">
           <string>View hits</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="button_view_stream">
          <property name="text">
           <string>View stream</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="button_sum_blanks">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>120</width>
            <height>25</height>
           </size>
          </property>
          <property name="text">
           <string>Sum of blanks</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="button_sum_hits">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>120</width>
            <height>25</height>
           </size>
          </property>
          <property name="text">
           <string>Sum of hits</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="button_peak_powder">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>120</width>
            <height>25</height>
           </size>
          </property>
          <property name="text">
           <string>Peak powder</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="button_peakogram">
          <property name="enabled">
           <bool>true</bool>
          </property>
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="maximumSize">
           <size>
            <width>120</width>
            <height>25</height>
           </size>
          </property>
          <property name="text">
           <string>Peakogram</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacer">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTableWidget" name="table_status">
        <property name="lineWidth">
         <number>3</number>
        </property>
        <property name="verticalScrollBarPolicy">
         <enum>Qt::ScrollBarAsNeeded</enum>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOn</enum>
        </property>
        <property name="sortingEnabled">
         <bool>true</bool>
        </property>
        <property name="rowCount">
         <number>10</number>
        </property>
        <attribute name="horizontalHeaderCascadingSectionResizes">
         <bool>false</bool>
        </attribute>
        <attribute name="horizontalHeaderDefaultSectionSize">
         <number>90</number>
        </attribute>
        <attribute name="horizontalHeaderHighlightSections">
         <bool>false</bool>
        </attribute>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <row/>
        <column>
         <property name="text">
          <string>Run</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Dataset</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Rawdata</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Cheetah</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>H5Directory</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Nprocessed</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Nhits</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Hitrate%</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Recipe</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Calibration</string>
         </property>
        </column>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout2"/>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1200</width>
     <height>20</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="menu_file_command"/>
    <addaction name="menu_file_start_crawler"/>
    <addaction name="separator"/>
    <addaction name="menu_file_quit"/>
   </widget>
   <widget class="QMenu" name="menuCheetah">
    <property name="title">
     <string>Cheetah</string>
    </property>
    <widget class="QMenu" name="menuProcess_selected_runs">
     <property name="title">
      <string>Process selected runs</string>
     </property>
     <addaction name="separator"/>
     <addaction name="menu_cheetah_process_runs"/>
     <addaction name="menu_cheetah_process_streaming"/>
    </widget>
    <addaction name="menuProcess_selected_runs"/>
    <addaction name="menu_cheetah_kill_processing"/>
    <addaction name="menu_cheetah_process_jungfrau_darks"/>
   </widget>
   <widget class="QMenu" name="menuAnalysis">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="title">
     <string>Analysis</string>
    </property>
    <addaction name="menu_analysis_hitrate"/>
    <addaction name="menu_analysis_peakogram"/>
   </widget>
   <widget class="QMenu" name="menu_masks">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="title">
     <string>Masks</string>
    </property>
    <addaction name="menu_mask_maker"/>
    <addaction name="menu_mask_view"/>
    <addaction name="menu_mask_psana"/>
   </widget>
   <widget class="QMenu" name="menuPowder">
    <property name="title">
     <string>View</string>
    </property>
    <addaction name="menu_powder_peaks_hits"/>
    <addaction name="menu_powder_peaks_blanks"/>
    <addaction name="menu_powder_hits_sum"/>
    <addaction name="menu_powder_blanks_sum"/>
   </widget>
   <widget class="QMenu" name="menuLogs">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="title">
     <string>Logs</string>
    </property>
    <addaction name="menu_log_batch"/>
    <addaction name="menu_log_cheetah_status"/>
   </widget>
   <widget class="QMenu" name="menuIndexing">
    <property name="title">
     <string>Indexing</string>
    </property>
    <addaction name="menu_indexing_view_stream"/>
    <addaction name="menu_indexing_cell_explorer"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuCheetah"/>
   <addaction name="menu_masks"/>
   <addaction name="menuPowder"/>
   <addaction name="menuAnalysis"/>
   <addaction name="menuIndexing"/>
   <addaction name="menuLogs"/>
  </widget>
  <action name="menu_file_autorefresh">
   <property name="text">
    <string>Auto refresh table</string>
   </property>
  </action>
  <action name="menu_file_refreshtable">
   <property name="text">
    <string>Refresh table</string>
   </property>
  </action>
  <action name="menu_file_startcrawler">
   <property name="text">
    <string>Start crawler</string>
   </property>
  </action>
  <action name="menu_file_command">
   <property name="text">
    <string>Unlock command operations</string>
   </property>
  </action>
  <action name="menu_file_quit">
   <property name="text">
    <string>Quit</string>
   </property>
  </action>
  <action name="menu_cheetah_autorun">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Autorun when new data is ready</string>
   </property>
  </action>
  <action name="menu_mask_combine">
   <property name="text">
    <string>Combine masks</string>
   </property>
  </action>
  <action name="menu_analysis_peakogram">
   <property name="text">
    <string>Peakogram</string>
   </property>
  </action>
  <action name="menu_analysis_hitrate">
   <property name="text">
    <string>Hitrate plot</string>
   </property>
  </action>
  <action name="menu_powder_hits_sum">
   <property name="text">
    <string>Virtual powder hits (class 1, detector corrected ADU sum)</string>
   </property>
  </action>
  <action name="menu_powder_blanks_sum">
   <property name="text">
    <string>Virtual powder blanks (class 0, detector corrected ADU sum)</string>
   </property>
  </action>
  <action name="menu_powder_peaks_hits">
   <property name="text">
    <string>Peakfinder virtual powder hits (class 1)</string>
   </property>
  </action>
  <action name="menu_powder_peaks_blanks">
   <property name="text">
    <string>Peakfinder virtual powder blanks (class 0)</string>
   </property>
  </action>
  <action name="menu_log_batch">
   <property name="text">
    <string>Batch log file</string>
   </property>
  </action>
  <action name="menu_log_cheetah">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Cheetah log file</string>
   </property>
  </action>
  <action name="menu_log_cheetah_status">
   <property name="enabled">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Cheetah status file</string>
   </property>
  </action>
  <action name="menu_mask_maker">
   <property name="text">
    <string>Mask maker</string>
   </property>
  </action>
  <action name="menu_mask_view">
   <property name="text">
    <string>View mask file</string>
   </property>
  </action>
  <action name="menu_powder_copydarkcal">
   <property name="text">
    <string>Copy darkcal.h5 to calib/darkcal</string>
   </property>
  </action>
  <action name="menu_file_modify_beamline_configuration">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Modify beamline configuration</string>
   </property>
  </action>
  <action name="menu_modify_config_files">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Modify config files</string>
   </property>
  </action>
  <action name="menu_file_start_crawler">
   <property name="text">
    <string>Start crawler</string>
   </property>
  </action>
  <action name="menu_cheetah_kill_processing">
   <property name="text">
    <string>Kill processing jobs</string>
   </property>
  </action>
  <action name="menu_cheetah_process_jungfrau_darks">
   <property name="text">
    <string>Process JF 1M darks</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
  </action>
  <action name="menu_mask_psana">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Extract mask from psana</string>
   </property>
   <property name="visible">
    <bool>false</bool>
   </property>
  </action>
  <action name="menu_indexing_view_stream">
   <property name="text">
    <string>View stream file</string>
   </property>
  </action>
  <action name="menu_indexing_cell_explorer">
   <property name="text">
    <string>Cell explorer</string>
   </property>
  </action>
  <action name="menu_cheetah_process_runs">
   <property name="text">
    <string>Save hits to files</string>
   </property>
  </action>
  <action name="menu_cheetah_process_streaming">
   <property name="text">
    <string>Stream hits to CrystFEL</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
